### PR TITLE
Remove 'clean' from build instructions

### PR DESCRIPTION
**Pull request by oehme**
## Overview

Gradle has excellent incremental build support. Invoking 'clean'
is a big waste of the developers time. As a rule of thumb, when
the developer has to invoke 'clean' to make the build work,
 the build is broken.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

**Comment by marcphilipp** [comment]
I agree with your point in general. IIRC you should use a `clean` after using Clover, though.

@junit-team/junit-lambda What are your opinions on this one?

**Comment by sbrannen** [comment]
If you're running the tests frequently via Gradle, then I agree that `gradle test` is sufficient.

However, if you're using `cloverHtmlReport`, `assemble`, or `install`, I would then tend to prefer `gradle clean ...` just to be safe.

More importantly, the use of `clean` in the instructions is intended for the average user when executing _pre-commit checks_, again just to make sure things are totally in order before submitting a PR, etc.

It's assumed (albeit perhaps incorrectly) that experienced Gradle users know when it makes sense to clean and when it's ok to skip that.

In summary, I'd leave `clean` in most (if not all) examples and perhaps add a note regarding omitting `clean` for subsequent, frequent test runs.

**Comment by oehme** [comment]
> However, if you're using cloverHtmlReport, assemble, or install, I would then tend to prefer gradle clean ... just to be safe.

This might be leftover fear from using Maven too long ;) Seriously, this is not necessary with Gradle. Gradle knows the inputs and outputs of your tasks, so there is no need to clean between invocations.

> It's assumed (albeit perhaps incorrectly) that experienced Gradle users know when it makes sense to clean and when it's ok to skip that.

You don't need to be experienced. If the build doesn't work and `clean` fixes it, then the build is broken. Usually due to a task not being implemented correctly.

> More importantly, the use of clean in the instructions is intended for the average user when executing pre-commit checks, again just to make sure things are totally in order before submitting a PR, etc.

I know I'm repeating myself, but: This will work just fine with incremental builds.

**Comment by marcphilipp** [comment]
Team decision:

- Keep `clean` in README
- Document Gradle's incremental build support and that you can usually omit `clean`

**Comment by oehme** [comment]
What were the reasons that lead to this decision? Is there something the Gradle team can do?

**Comment by marcphilipp** [comment]
Maybe we're just old-fashioned developers who have used Maven for too long. ;-)
